src/test/txvalidationcache_tests.cpp

```diff
@@ -105,6 +105,8 @@ BOOST_FIXTURE_TEST_CASE(tx_mempool_block_doublespend, TestChain100Setup)
 static void ValidateCheckInputsForAllFlags(CMutableTransaction &tx, uint32_t failing_flags, bool add_to_cache)
 {
     PrecomputedTransactionData txdata(tx);
+    CTransaction cTx(tx);
```

Comment:
in commit 384bcd116c7bdfbe7ed851a739a8d8c8809d4d12 ( speed up of tx_validationcache_tests by reusing of CTransaction), it seems you can instead just adjust the signature:

```diff
-static void ValidateCheckInputsForAllFlags(CMutableTransaction &tx, uint32_t failing_flags, bool add_to_cache)
+static void ValidateCheckInputsForAllFlags(const CTransaction &tx, uint32_t failing_flags, bool add_to_cache)
```

(Might need a rebase on master)

Also, to I suggest splitting up the commits into separate pull request to make it easier to review and merge the changes independently. (You could start with this change, since it seems a clear and trivial win)